be/src/column/column_helper.h

```diff
@@ -304,24 +304,26 @@ class ColumnHelper {
     }
 
     static Column* get_data_column(Column* column) {
-        if (column->is_nullable()) {
-            auto* nullable_column = down_cast<NullableColumn*>(column);
-            return nullable_column->mutable_data_column();
-        } else if (column->is_constant()) {
+        // Since const null is actually a const column, we need to handle const condition first.
+        if (column->is_constant()) {
             auto* const_column = down_cast<ConstColumn*>(column);
             return const_column->mutable_data_column()->get();
+        } else if (column->is_nullable()) {
+            auto* nullable_column = down_cast<NullableColumn*>(column);
+            return nullable_column->mutable_data_column();
         } else {
             return column;
         }
     }
 
     static const Column* get_data_column(const Column* column) {
-        if (column->is_nullable()) {
-            auto* nullable_column = down_cast<const NullableColumn*>(column);
-            return nullable_column->data_column().get();
-        } else if (column->is_constant()) {
+        // Since const null is actually a const column, we need to handle const condition first.
+        if (column->is_constant()) {
             auto* const_column = down_cast<const ConstColumn*>(column);
             return const_column->data_column().get();
```

Comment:
If the input column is const-null, what is data_column's type? Why is it not NullableColumn?

Reply:
As far as I know, const-null column should be handled by the caller.
This function can not accept this kind of column

Reply:
It's by design, const null must be a ConstColumn with a null value. 

```cpp
ColumnPtr ColumnHelper::create_const_null_column(size_t chunk_size) {
    auto nullable_column = NullableColumn::create(Int8Column::create(), NullColumn::create());
    nullable_column->append_nulls(1);
    return ConstColumn::create(nullable_column, chunk_size);
}
```

Reply:
Really appreciate the comment, it has been reverted.